plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1426,6 +1429,10 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
             transactionLogWriter.flush();
             writeCommitted = true;
 
+            if (handle.replace() && readVersion != commitVersion) {
+                writeCheckpointIfNeeded(session, schemaTableName, handle.location(), readVersion, handle.checkpointInterval(), commitVersion);
```

Comment:
Add a similar change in `io.trino.plugin.deltalake.DeltaLakeMetadata#createTable` for `CREATE OR REPLACE TABLE mytable(data integer)` kind of statements. Add also a corresponding test.

Reply:
Adding this logic and coverage on `TestDeltaLakeFileOperations#testCreateOrReplaceTable`